src/Erl/Test/EUnit.purs

```diff
@@ -70,17 +73,29 @@ test l t = liftF $ TestUnit l t unit
 empty :: TestSuite
 empty = liftF $ TestEmpty unit
 
-setupTeardown :: Setup -> Teardown -> TestSuite -> TestSuite
-setupTeardown s t su = liftF $ TestState s t su unit
+testDataToForeign :: forall a. a -> Foreign
+testDataToForeign = unsafeCoerce
+
+teardownToForeign :: forall a. TestTeardown a -> Teardown
+teardownToForeign = unsafeCoerce
 
-setup :: Setup -> TestSuite -> TestSuite
-setup s su = liftF $ TestState s (pure unit) su unit
+testSuiteToForeign :: forall a. (a -> TestSuite) -> (Foreign -> TestSuite)
+testSuiteToForeign = unsafeCoerce
 
-teardown :: Teardown -> TestSuite -> TestSuite
-teardown t su = liftF $ TestState (pure unit) t su unit
+setupTeardown :: forall a. TestSetup a -> TestTeardown a -> (a -> TestSuite) -> TestSuite
+setupTeardown s t su = liftF $ TestState (map testDataToForeign s) (teardownToForeign t) (testSuiteToForeign su) unit
```

Comment:
Was wondering if we should call this one `setupTeardown'` so we can keep around the current implementation for when you want to ignore the runtime data

Reply:
if anything just add `setupTeardown_` for the old one to ignore

Reply:
Done